sdk/eventhub/azure-eventhub/tests/perfstress_tests/_test_base.py

```diff
@@ -0,0 +1,75 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure_devtools.perfstress_tests import PerfStressTest, get_random_bytes
+from azure.eventhub import EventHubProducerClient, EventHubConsumerClient, EventData
+from azure.eventhub.aio import (
+    EventHubProducerClient as AsyncEventHubProducerClient,
+    EventHubConsumerClient as AsyncEventHubConsumerClient
+)
+
+
+class _EventHubTest(PerfStressTest):
+    consumer_group = '$Default'
+
+    @staticmethod
+    def add_arguments(parser):
+        super(_EventHubTest, _EventHubTest).add_arguments(parser)
+        parser.add_argument('--event-size', nargs='?', type=int, help='Size of a single event. Defaults to 100 bytes', default=100)
+        parser.add_argument('--num-events', nargs='?', type=int, help='Number of events to send or receive. Defaults to 100', default=100)
+
+
+class _SendTest(_EventHubTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        connection_string = self.get_from_env("AZURE_EVENTHUB_CONNECTION_STRING")
+        eventhub_name = self.get_from_env("AZURE_EVENTHUB_NAME")
+        self.producer = EventHubProducerClient.from_connection_string(connection_string, eventhub_name=eventhub_name)
+        self.async_producer = AsyncEventHubProducerClient.from_connection_string(connection_string, eventhub_name=eventhub_name)
+
+    async def close(self):
+        self.producer.close()
+        await self.async_producer.close()
+        await super().close()
+
+
+class _ReceiveTest(_EventHubTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        connection_string = self.get_from_env("AZURE_EVENTHUB_CONNECTION_STRING")
+        eventhub_name = self.get_from_env("AZURE_EVENTHUB_NAME")
+        self.async_producer = AsyncEventHubProducerClient.from_connection_string(connection_string, eventhub_name=eventhub_name)
```

Comment:
qq: I'm assuming you're only using the async producer, rather than sync too, to send events since the consumers are what is being tested here. If you used only the sync producer, would it accomplish the same thing? (for my own curiosity)

Reply:
yes! the producer is just for preload the eventhub with events, so either sync or async producer would work.

Reply:
Maybe we could name it accordingly - to avoid confusing it with the testable clients. Something like `self.event_preloader`